src/Login.js

```diff
@@ -8,40 +8,48 @@ import {
   FormControl,
   HelpBlock,
   Button,
-  Alert
+  Alert,
+  Form
 } from 'react-bootstrap'
 import { withRouter } from 'react-router'
 
 import NavBar from './components/NavBar'
 import { Api } from './utils'
-
-function FieldGroup({ id, label, help, ...props }) {
-  return (
-    <FormGroup controlId={id}>
-      <ControlLabel>{label}</ControlLabel>
-      <FormControl {...props} />
-      {help && <HelpBlock>{help}</HelpBlock>}
-    </FormGroup>
-  )
-}
+import { FieldGroup } from './components'
 
 class Login extends Component {
-  state = {
-    error: null
+  constructor(props) {
+    super(props)
+    this.state = {
+      error: null,
+      username: '',
+      password: ''
+    }
+    this.onChangeUsername = this.onChangeUsername.bind(this)
+    this.onChangePassword = this.onChangePassword.bind(this)
+    this.onSubmit = this.onSubmit.bind(this)
+  }
+
+  onChangeUsername(inputUsername) {
+    this.setState({
+      username: inputUsername.target.value
+    })
+  }
+
+  onChangePassword(inputPassword) {
+    this.setState({
+      password: inputPassword.target.value
+    })
   }
 
-  onSubmitForm = e => {
-    e.preventDefault()
-    const { username, password } = e.target
+  onSubmit() {
+    const { username, password } = this.state
     const data = {
-      username: username.value,
-      password: password.value
+      username: username,
+      password: password
```

Comment:
You can use a shorthand notation for this since the key and value have the same name:

```javascript
const data = {
  username,
  password
}
```